src/api/evaluators/custom_evals/eval_combined.py

```diff
@@ -0,0 +1,333 @@
+import json
+import os
+from datetime import timedelta
+from typing import Any, Dict, List
+from azure.core.exceptions import HttpResponseError
+from azure.identity import DefaultAzureCredential
+from azure.monitor.query import LogsQueryClient, LogsQueryStatus
+from dotenv import load_dotenv
+import prompty
+import prompty.azure
+import logging
+
+# Configure logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
+
+# Load environment variables
+load_dotenv()
+
+
+
+# Configuration for the Azure OpenAI endpoint
+model_config = {
+    "azure_endpoint": os.environ["AZURE_OPENAI_ENDPOINT"],
+    "api_version": os.environ["AZURE_OPENAI_API_VERSION"],
+}
+
+
+# Authenticate using DefaultAzureCredential
+credential = DefaultAzureCredential()
+logs_client = LogsQueryClient(credential)
+workspace_id= os.environ["APPINSIGHTS_WORKSPACE_ID"]
+
+
+def extract_app_insights_data( timespan_days: int = 2, take_count: int = 5) -> List[Dict[str, Any]]:
+    """
+    Extracts data from Azure Application Insights for multiple interactions.
+
+    Args:
+        workspace_id (str): The workspace ID to query.
+        timespan_days (int, optional): Number of days for the query timespan. Defaults to 2.
+        take_count (int, optional): Number of interactions to retrieve. Defaults to 5.
+
+    Returns:
+        List[Dict[str, Any]]: A list of dictionaries containing the extracted data.
+    """
+    try:
+        # Authenticate using DefaultAzureCredential
+        credential = DefaultAzureCredential()
```

Comment:
What's the difference between the credential here and the one on line 30?

Reply:
fixed.